### PR TITLE
[Merged by Bors] - feat(Algebra/GroupPower): weaken TC assumptions

### PR DESCRIPTION
**Pull request by urkud**
- Assume `OrderedSemiring` instead of `StrictOrderedSemiring`
  here and there.
- Add `@[simp]` to `zsmul_one`.
- Add `exists_lt_nsmul`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by alexjbest** [comment]
maintainer merge

File: Mathlib/Algebra/GroupPower/Lemmas.lean [hunk 1]
```diff
@@ -645,13 +646,13 @@ theorem one_add_mul_le_pow' (Hsq : 0 ≤ a * a) (Hsq' : 0 ≤ (1 + a) * (1 + a))
...
     calc
       1 + (↑(n + 2) : R) * a ≤ 1 + ↑(n + 2) * a + (n * (a * a * (2 + a)) + a * a) :=
-        (le_add_iff_nonneg_right _).2 this
+        le_add_of_nonneg_right this
       _ = (1 + a) * (1 + a) * (1 + n * a) := by {
           simp only [Nat.cast_add, add_mul, mul_add, one_mul, mul_one, ← one_add_one_eq_two,
             Nat.cast_one, add_assoc, add_right_inj]
           simp only [← add_assoc, add_comm _ (↑n * a)]
-          simp only [add_assoc, add_right_inj, (n.cast_commute (_ : R)).left_comm]
-          ac_rfl }
```
Comment (alexjbest) [inline Mathlib/Algebra/GroupPower/Lemmas.lean:654]: did the ac_rfl stop working? it seems a more principled and maintainable proof than simp only to me

File: Mathlib/Algebra/GroupPower/Lemmas.lean [hunk 2]
```diff
@@ -630,9 +631,9 @@ theorem neg_one_pow_eq_pow_mod_two [Ring R] {n : ℕ} : (-1 : R) ^ n = (-1) ^ (n
   rw [← Nat.mod_add_div n 2, pow_add, pow_mul]; simp [sq]
 #align neg_one_pow_eq_pow_mod_two neg_one_pow_eq_pow_mod_two
 
-section StrictOrderedSemiring
+section OrderedSemiring
 
-variable [StrictOrderedSemiring R] {a : R}
+variable [OrderedSemiring R] {a : R}
```
Comment (eric-wieser) [inline Mathlib/Algebra/GroupPower/Lemmas.lean:636]: I'm a little wary of this one: didn't we decide `OrderedSemiring` was nonsense? Maybe @YaelDillies remembers the history better.

Comment (Vierkantor) [reply Mathlib/Algebra/GroupPower/Lemmas.lean:636]: I believe Yaël fixed the weirdness a while ago: [mathlib#16172](https://github.com/leanprover-community/mathlib/pull/16172).

**Review by Vierkantor — APPROVED** [review]
LGTM with Alex's comments. I'll let Eric decide if `OrderedSemiring` is non-nonsense enough to get this merged :)

bors d=@eric-wieser

File: Mathlib/Algebra/GroupPower/Lemmas.lean (hunk 2 above)
Comment (Vierkantor) [reply Mathlib/Algebra/GroupPower/Lemmas.lean:636]: By the way, Yaël is on vacation this week, according to Zulip. So I don't want to block on them :)

**Comment by eric-wieser** [comment]
Thanks for the archaeology @Vierkantor, I'm convinced!

bors d+

(we're waiting to address Alex's comment still, right?)

File: Mathlib/Algebra/GroupPower/Lemmas.lean (hunk 1 above)
Comment (urkud) [reply Mathlib/Algebra/GroupPower/Lemmas.lean:654]: Yes, it didn't work. I don't know why.

Comment (urkud) [reply Mathlib/Algebra/GroupPower/Lemmas.lean:654]: @alexjbest Do you want to have a look, or should I merge it?

Comment (alexjbest) [reply Mathlib/Algebra/GroupPower/Lemmas.lean:654]: I'd like to understand why but it shouldn't hold up this PR! so please merge and I can investigate later

**Comment by urkud** [comment]
bors merge

**Comment by urkud** [comment]
Both failures are because of `lake exe cache get`
bors merge

File: Mathlib/Algebra/GroupPower/Lemmas.lean (hunk 1 above)
Comment (alexjbest) [reply Mathlib/Algebra/GroupPower/Lemmas.lean:654]: minimised and reported as lean4#2413

File: Mathlib/Algebra/GroupPower/Lemmas.lean (hunk 2 above)
Comment (YaelDillies) [reply Mathlib/Algebra/GroupPower/Lemmas.lean:636]: Just came back from my holidays. Anne is perfectly right!

**Pull request closed without merging**